dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts

```diff
@@ -176,7 +180,11 @@ const getUserConfigs = (
             [GROUPBY]: [
               {
                 childField: { ...(axesData.xaxis ? axesData.xaxis[0] : initialEntryTreemap) },
-                parentFields: [],
+                parentFields:
+                  userSelectedConfigs?.dataConfig !== undefined &&
+                  userSelectedConfigs?.dataConfig[GROUPBY]?.length > 0
```

Comment:
The question mark after `userSelectedConfigs` may not be required here, as the first expression before && is going to be evaluated first. If that becomes true, then the property `dataConfig` does exist anyway.